### PR TITLE
test: add unit tests for notification_routing (Fixes #832)

### LINKED ISSUE #832: test : add unit tests for notification_routing
Summary of What Needs to be Done:
The notification_routing service handles routing notifications but has no test file. Need tests for routing rules and channel selection.

Changes that Need to be Made:
- Create backend/tests/test_notification_routing.py
- Add tests for core functionality
- Test edge cases and error handling

Impact that it would Provide:
- Improve test coverage for services
- Prevent regression in service functionality
- Verify service logic works correctly

Please assign this issue to me so I can implement it.

### PR DESCRIPTION
**Pull request by zeroknowledge0x**
## Summary
Adds a comprehensive, isolated unit test suite for `NotificationRoutingMiddleware`.

## Changes
- Creates `backend/tests/test_notification_routing.py`
- Covers `NotificationType` enum contracts
- Covers system settings retrieval success, missing data, defaults, and fail-open error behavior
- Covers per-company settings cache behavior and cache invalidation
- Covers email, admin alert, and push notification gating rules
- Covers notification logging helpers and singleton lifecycle (`load` / `get_instance`)

## Testing
```bash
pytest backend/tests/test_notification_routing.py -q
```
Result: `29 passed in 2.69s`.

## Related Issues
Fixes #832

## Notes
This PR intentionally targets `notification_routing.py`. The earlier PR body mistakenly linked #824, which asks for tests around a different `notification_service.py` module that is not present in the current repository tree.

### CODE REVIEW

File: backend/tests/test_notification_routing.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+Tests cover:
+- NotificationType enum values
+- Settings fetching (success, failure, caching)
+- Email notification gating (enabled, disabled, digest frequency)
+- Admin alert gating
+- Push notification gating
+- Cache invalidation
+- Singleton pattern (load / get_instance)
+- Logging behavior
+"""
```
[the remark this answers is not in the document]
Comment (zeroknowledge0x) [reply backend/tests/test_notification_routing.py:13]: Thanks for the catch — verified against the current repo and the finding was valid for the original PR metadata, not the code. The added tests are for `NotificationRoutingMiddleware` / `notification_routing.py`, while #824 asks for a different `notification_service.py` module that is not present in the current tree.

Fixed by updating the PR title/body to target the matching routing test issue instead: `Fixes #832`.

Validation run:
```bash
pytest backend/tests/test_notification_routing.py -q
# 29 passed in 2.69s
```

**Comment by zeroknowledge0x** [comment]
Payment / bounty info:

- GitHub: @zeroknowledge0x
- EVM wallet (USDC/ETH): 0x036E525f4569865DA76A0ef758304b309740aeef
- Ready to receive payout when this gets merged/reviewed.

Thanks!

**Comment by riteshbonthalakoti** [comment]
Hi @zeroknowledge0x! 🙌

Thank you so much for your excellent contribution: **"test: add unit tests for notification_routing (Fixes #832)"**! We really appreciate the high-quality code and effort you have put into the platform.

Just a quick, friendly heads-up as we prepare our manual merging and verification queues—please make sure to complete all the mandatory community steps listed below. 

Once those manual steps are verified, we'll get your PR officially merged into the `gssoc` branch (or keep it neatly cataloged if closed as integrated) and assign it the highest possible GSSoC S-Tier labels to maximize your leaderboard points!

Let's build something amazing together! 🚀🔥


---

### 🌟 Community Support & Network Steps (Take 10 Seconds!)
As we prepare our manual verification and merging queues, please make sure you have taken a moment to complete these required steps to finalize your points:
1. ⭐ **Star this repository**: https://github.com/ritesh-1918/HELPDESK.AI (Mandatory)
2. 🍴 **Fork this repository**: https://github.com/ritesh-1918/HELPDESK.AI/fork (Mandatory)
3. 👤 **Follow @ritesh-1918 on GitHub**: https://github.com/ritesh-1918 (Mandatory - manual step)
4. 💼 **Connect on LinkedIn**: https://www.linkedin.com/in/ritesh1908/ (Mandatory)

*Note: Having these steps completed manually is required before your PR points are officially cleared.*

**Comment by riteshbonthalakoti** [comment]
Hi @zeroknowledge0x! 🙌

Thank you so much for your excellent contribution: **"test: add unit tests for notification_routing (Fixes #832)"**! We really appreciate the high-quality code and effort you have put into the platform.

Just a quick, friendly heads-up as we prepare our manual merging and verification queues—please make sure to complete all the mandatory community steps listed below. 

Once those manual steps are verified, we'll get your PR officially merged into the `gssoc` branch (or keep it neatly cataloged if closed as integrated) and assign it the highest possible GSSoC S-Tier labels to maximize your leaderboard points!

Let's build something amazing together! 🚀🔥


---

### 🌟 Community Support & Network Steps (Take 10 Seconds!)
As we prepare our manual verification and merging queues, please make sure you have taken a moment to complete these required steps to finalize your points:
1. ⭐ **Star this repository**: https://github.com/ritesh-1918/HELPDESK.AI (Mandatory)
2. 🍴 **Fork this repository**: https://github.com/ritesh-1918/HELPDESK.AI/fork (Mandatory)
3. 👤 **Follow @ritesh-1918 on GitHub**: https://github.com/ritesh-1918 (Mandatory - manual step)
4. 💼 **Connect on LinkedIn**: https://www.linkedin.com/in/ritesh1908/ (Mandatory)

*Note: Having these steps completed manually is required before your PR points are officially cleared.*

**Comment by riteshbonthalakoti** [comment]
Hi @zeroknowledge0x! 🙌

Thank you so much for your excellent contribution: **"test: add unit tests for notification_routing (Fixes #832)"**! We really appreciate the high-quality code and effort you have put into the platform.

Just a quick, friendly heads-up as we prepare our manual merging and verification queues—please make sure to complete all the mandatory community steps listed below. 

⚠️ **Quick leaderboard tip**: To secure the absolute highest GSSoC point tier for your contribution, please make sure you are following our project admin @ritesh-1918 manually on GitHub. Having all four onboarding steps cleared is highly preferable for manual PR approvals! ✨

Once those manual steps are verified, we'll get your PR officially merged into the `gssoc` branch (or keep it neatly cataloged if closed as integrated) and assign it the highest possible GSSoC S-Tier labels to maximize your leaderboard points!

Let's build something amazing together! 🚀🔥


---

### 🌟 Project Support & Developer Network (Show Some Love!)
As we prepare our manual verification and merging queues, please take a quick moment to ensure you have completed all four community steps:
1. ⭐ **Star this repository**: Helps our AI helpdesk get noticed! [Star the repo here](https://github.com/ritesh-1918/HELPDESK.AI)
2. 🍴 **Fork this repository**: Keep a copy to build your own cool tools! [Fork here](https://github.com/ritesh-1918/HELPDESK.AI/fork)
3. 👤 **Follow @ritesh-1918 on GitHub**: Stay updated on real-time open-source projects! [Follow ritesh-1918 here](https://github.com/ritesh-1918)
4. 💼 **Connect on LinkedIn**: Let's build a strong engineering connection! [Connect with Ritesh on LinkedIn](https://www.linkedin.com/in/ritesh1908/)

*Note: Having all four steps completed manually is required before your PR points are officially cleared.*
